flask_backend_design.py

```diff
@@ -0,0 +1,272 @@
+'"'TransactionDate'": '+CONVERT(NVARCHAR(20),TransactionDate) TransactionDate\
+    FROM Transactions_bankdemo WITH(Nolock) \
+    where accountnumber=? \
+    AND CONVERT(DATE,TransactionDate)=CONVERT(DATE,GETDATE())",
+                       accountnumber)
+        rows = cursor.fetchall()
+        a = []
+        if rows:
+            for row in rows:
+                a.append(list(row))
+            return jsonify({"results": a})
+        else:
+            return "No Transactions for this account"
+    if date_range == '1 Week':
+        cursor.execute("SELECT \
+'"'accountnumber'": '+convert(NVARCHAR(20),accountnumber) As accountnumber,\
+'"'TransactionType'": '+TransactionType TransactionType,\
+'"'Amount'": '+CONVERT(NVARCHAR(20),CONVERT(float,Amount)) Amount,\
+'"'TransactionDate'": '+CONVERT(NVARCHAR(20),TransactionDate) TransactionDate \
+        FROM Transactions_bankdemo WITH(Nolock)  \
+        where accountnumber=? \
+AND DATEDIFF(DD,CONVERT(DATE,TransactionDate),CONVERT(DATE,GETDATE()))< 7",
+                       accountnumber)
+        rows = cursor.fetchall()
+        a = []
+        if rows:
+            for row in rows:
+                a.append(list(row))
+            return jsonify({"results": a})
+        else:
+            return "No Transactions for this account"
+    if date_range == '1 Month':
+        cursor.execute("SELECT \
+'"'accountnumber'": '+convert(NVARCHAR(20),accountnumber) As accountnumber,\
+'"'TransactionType'": '+TransactionType TransactionType,\
+'"'Amount'": '+CONVERT(NVARCHAR(20),CONVERT(float,Amount)) Amount,\
+'"'TransactionDate'": '+CONVERT(NVARCHAR(20),TransactionDate) TransactionDate \
+        FROM Transactions_bankdemo WITH(Nolock)  \
+        where accountnumber=? \
+AND DATEDIFF(Month,CONVERT(DATE,GETDATE()),CONVERT(DATE,TransactionDate))< 1",
+                       accountnumber)
+        rows = cursor.fetchall()
+        a = []
+        if rows:
+            for row in rows:
+                a.append(list(row))
+            return jsonify({"results": a})
+        else:
+            return "No Transactions for this account"
+    if date_range == '1 Year':
+        cursor.execute("SELECT \
+'"'accountnumber'": '+convert(NVARCHAR(20),accountnumber) As accountnumber,\
+'"'TransactionType'": '+TransactionType TransactionType,\
+'"'Amount'": '+CONVERT(NVARCHAR(20),CONVERT(float,Amount)) Amount,\
+'"'TransactionDate'": '+CONVERT(NVARCHAR(20),TransactionDate) TransactionDate \
+        FROM Transactions_bankdemo WITH(Nolock)  \
+        where accountnumber=? \
+AND DATEDIFF(Year,CONVERT(DATE,GETDATE()),CONVERT(DATE,TransactionDate))< 1",
+                       accountnumber)
+        rows = cursor.fetchall()
+        a = []
+        if rows:
+            for row in rows:
+                a.append(list(row))
+            return jsonify({"results": a})
+        else:
+            return "No Transactions for this account"
+
+
+app.run(debug=True)
+
+
+
+
```

Comment:
Lots of blank lines at the end.

Code is not pep8 and pylint compliant.

Reply:
I do not see any CRUD APIs.
How to create, update, read or delete an account or customer.

Reply:
Practice code or non related code should not be part of same PR containing main feature code.
This can never be approved and merged because merging this PR will also merge unwanted code to master.

Reply:
Where is flask app code. app.run ?